### PR TITLE
Docs Cloudbuild Webhook Trigger

### PR DESCRIPTION
**Pull request by PeyGis**
# Description

This doc guides users to trigger webhook actions to Google CloudBuild

## Added docs pages

docs/create-self-service-experiences/setup-backend/cloudbuild-pipeline/

### CODE REVIEW

File: docs/create-self-service-experiences/setup-backend/cloudbuild-pipeline/cloudbuild-pipeline.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+:::
+
+### Setting up the Port action
+
+To trigger the Cloud Build pipeline, you will setup a Port [Webhook Action](../webhook/).
+
+Here is an example for an action that will trigger the webhook you just set up:
+
+```json showLineNumbers
+[
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/cloudbuild-pipeline/cloudbuild-pipeline.md:178]: Put this through a proper JSON formatter, you can see that the root square bracket and curly bracket are in the same indent and it doesn't look good enough

File: docs/create-self-service-experiences/setup-backend/cloudbuild-pipeline/cloudbuild-pipeline.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+{
+  "identifier": "myIdentifier",
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/cloudbuild-pipeline/cloudbuild-pipeline.md:180]:
```suggestion
  "identifier": "runPipeline",
```

File: docs/create-self-service-experiences/setup-backend/cloudbuild-pipeline/cloudbuild-pipeline.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+  "title": "GCP Cloud Build",
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/cloudbuild-pipeline/cloudbuild-pipeline.md:181]:
```suggestion
  "title": "Run GCP Cloud Build Pipeline",
```

File: docs/create-self-service-experiences/setup-backend/cloudbuild-pipeline/cloudbuild-pipeline.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+  "icon": "GCP",
+  "userInputs": {
+    "properties": {
+      "region": {
+        "type": "string",
+        "title": "Region Name"
+      },
+      "image_name": {
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/cloudbuild-pipeline/cloudbuild-pipeline.md:189]:
```suggestion
      "imageName": {
```

File: docs/create-self-service-experiences/setup-backend/cloudbuild-pipeline/cloudbuild-pipeline.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+        "${_REGION}",
+        "--platform",
+        "managed",
+        "--allow-unauthenticated",
+      ]
+
+substitutions:
+  #Repository Specific configuration. DevOps can change this settings
+  _REPOSITORY: your_container_name
+  _IMAGE_NAME: $(body.payload.properties.image_name)
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/cloudbuild-pipeline/cloudbuild-pipeline.md:89]:
```suggestion
  _IMAGE_NAME: $(body.payload.properties.imageName)
```

File: docs/create-self-service-experiences/setup-backend/cloudbuild-pipeline/cloudbuild-pipeline.md [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+
+And finally, you need to connect your source code repository to this pipeline and specify whether you want to invoke the pipeline steps in your `cloudbuild.yaml` or `Dockerfile`.
+
+![Webhook Security](../../../../static/img/self-service-actions/setup-backend/cloudbuild-pipeline/webhook_repository.PNG)
+
+### Defining variables
+
+Google Cloud Build enables developers to bind incoming webhook payload data to the pipeline configuration file. If you take a look at the sample `cloudbuild.yaml` file above, you will see the **substitutions** section has the below content. This is where you define the variables which will be passed to your pipeline run.
+
+```yaml
```
[the remark this answers is not in the document]
Comment (MPTG94) [reply docs/create-self-service-experiences/setup-backend/cloudbuild-pipeline/cloudbuild-pipeline.md:114]: Please verify all codeblocks in this doc have `showLineNumbers`

**Review by MPTG94 — CHANGES_REQUESTED** [review]
Please fix the various comments

File: docs/create-self-service-experiences/setup-backend/cloudbuild-pipeline/_category_.json [hunk 7]
```diff
@@ -0,0 +1,4 @@
+{
+  "label": "Cloud Build Pipeline Trigger",
+  "position": 6
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/cloudbuild-pipeline/_category_.json:3]: Azure pipelines already uses position `6`, if you want to add it after Azure, you need to change Kafka to `8` and cloud build will use `7`.

**Review by MPTG94 — COMMENTED** [review]
Added another comment

File: docs/create-self-service-experiences/setup-backend/cloudbuild-pipeline/_category_.json (hunk 7 above)
Comment (PeyGis) [reply docs/create-self-service-experiences/setup-backend/cloudbuild-pipeline/_category_.json:3]: Noted

File: docs/create-self-service-experiences/setup-backend/cloudbuild-pipeline/cloudbuild-pipeline.md (hunk 1 above)
Comment (PeyGis) [reply docs/create-self-service-experiences/setup-backend/cloudbuild-pipeline/cloudbuild-pipeline.md:178]: Thanks for pointing this out. I've fixed it

File: docs/create-self-service-experiences/setup-backend/cloudbuild-pipeline/cloudbuild-pipeline.md [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+}
+```
+
+For example, the substitutions value for the `region` input would be:
+
+```text showLineNumbers
+$.payload.properties.region
+```
+
+**Port Action** - The full port action definition can be found [here](#setting-up-the-port-action).
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/cloudbuild-pipeline/cloudbuild-pipeline.md:175]:
```suggestion
**Port Action** - The full Port action definition can be found [here](#setting-up-the-port-action).
```

File: static/img/self-service-actions/setup-backend/cloudbuild-pipeline/cloud-build-illustration.png:Zone.Identifier
Comment (MPTG94) [inline static/img/self-service-actions/setup-backend/cloudbuild-pipeline/cloud-build-illustration.png:Zone.Identifier:1]: @PeyGis remove this file

**Review by MPTG94 — CHANGES_REQUESTED** [review]
Requested some more minor changes

**Review by MPTG94 — APPROVED** [review]
LGTM

**Pull request merged**